src/main/java/kg/jarkyn/cobspecserver/ClientSocket.java

```diff
@@ -0,0 +1,28 @@
+package kg.jarkyn.cobspecserver;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class ClientSocket implements Client {
```

Comment:
Even if it was called something like SimpleSocket or something to suggest it is really just a socket, I would have understood it a bit quicker.

And given you'll probably only have one of these Sockets in your app, then an interface may not be necessary if it's just for creating test doubles, however this argument is probably up for debate. If you want a test double for this class, you can always subclass said class/override the required methods (not sure how others feel about that)..